saleor/order/utils.py

```diff
@@ -48,7 +54,36 @@ def order_status_change(sender, instance, **kwargs):
             logger.exception('Recording order in analytics failed')
 
 
+def cancel_order(order):
+    """Cancells order by cancelling all associated shipment groups."""
+    for group in order.groups.all():
+        cancel_delivery_group(group, cancel_order=False)
+    order.status = OrderStatus.CANCELLED
+    order.save()
+
+
+def recalculate_order(order):
```

Comment:
We have now two places with similar calculations - one is this function and the other is the `create_order` in `checkout/core`. They both calculate the total price and the shipping price. Maybe `recalculate_order` could be used there as well?

Reply:
It's quite different. ```recalculate_order``` calculates order's total price and saves it to an order. ```get_total``` in checkout behaves similar, but performs action on a list of partitions 
(even not delivery groups) of not existing order yet. It is used in few places, e .g. ```templates/checkout/details.html```. I'm not sure if merging them makes sense.